lib/Fmt_odoc.ml

```diff
@@ -14,7 +14,11 @@ open Odoc_parser.Ast
 module Loc = Odoc_parser.Loc
 
 type fmt_code =
-  Conf.t -> offset:int -> string -> (string, [`Msg of string]) Result.t
+     Conf.t
+  -> offset:int
+  -> set_margin:bool
```

Comment:
We might get rid of the `set_margin` by not using `Fmt.eval` when formatting code blocks in docstrings. That might require some time so I suggest to do it later and not to block this PR. The extra argument is not too bad as a temporary solution.